streams/src/main/java/org/apache/kafka/streams/kstream/Materialized.java

```diff
@@ -259,4 +285,16 @@ protected Materialized(final Materialized<K, V, S> materialized) {
         this.retention = retention;
         return this;
     }
+
+    /**
+     * Set the type of the materialized {@link StateStore}.
+     *
+     * @param storeType  the store type {@link StoreType} to use.
+     * @return itself
+     */
+    public Materialized<K, V, S> withStoreType(final StoreType storeType) throws IllegalArgumentException {
+        Objects.requireNonNull(storeType, "store type can't be null");
```

Comment:
Should we set `storeSupplier = null` here? Or throw an exception (or log a WARN)? It seems not to make sense to have both `storeType` and `storeSupplier` being set?

Reply:
Good suggestion! Updated and tests added.